File: webf/lib/src/dom/node_list.dart
/*
 * Copyright (C) 2024-present The OpenWebF Company. All rights reserved.
 * Licensed under GNU GPL with Enterprise exception.
 */
/*
 * Copyright (C) 2022-2024 The WebF authors. All rights reserved.
 */

import 'node.dart';


abstract class NodeList<T extends Node> extends Iterable<T> {
  // Constructor for NodeList
  NodeList();

  // DOM methods & attributes for NodeList
  bool get isEmptyNodeList => false;

  bool get isChildNodeList => false;

  Node get ownerNode;

  @override
  Iterator<T> get iterator;
}

File: bridge/core/dom/node_list.cc
/*
 * Copyright (C) 2019-2022 The Kraken authors. All rights reserved.
 * Copyright (C) 2022-present The WebF authors. All rights reserved.
 */

#include "node_list.h"

namespace webf {

void NodeList::InvalidateCache() {
  for (auto& cache : tag_collection_cache_)
    cache.second->InvalidateCache();
}

void NodeList::Trace(webf::GCVisitor* visitor) const {
  for (auto& item : tag_collection_cache_) {
    visitor->TraceMember(item.second);
  }
}

}  // namespace webf